apps/prairielearn/src/models/user.ts

```diff
@@ -17,6 +17,20 @@ export async function selectOptionalUserByUid(uid: string): Promise<User | null>
   return await queryOptionalRow(sql.select_user_by_uid, { uid }, UserSchema);
 }
 
+export async function selectUserByUidAndInstitution({
+  uid,
+  institution_id,
+}: {
+  uid: string;
+  institution_id: string;
+}): Promise<User | null> {
+  return await queryOptionalRow(
+    sql.select_user_by_uid_and_institution,
+    { uid, institution_id },
+    UserSchema,
+  );
+}
```

Comment:
Nit: UIDs are globally unique, so you might consider using the existing `selectOptionalUserByUid` function and then asserting that the institution ID matches.